test/browser/test.js

```diff
@@ -873,6 +873,34 @@ describe('morphdom' , function() {
         expect(el1.selectedIndex).to.equal(1);
     });
 
+    it('should update selection state on select elements when adding options', function () {
+        var el1 = node('select');
+
+        document.body.appendChild(el1);
+
+        var el2 = node('select');
+        el2.appendChild(node('option', {}, 'Option 1'));
+        el2.appendChild(node('option', {'selected': ''}, 'Option 2'));
+
+        morphdom(el1, el2);
+
+        expect(el1.selectedIndex).to.equal(1);
+    });
+
+    it('should select first option on select elements when no option explicitly selected', function () {
```

Comment:
I'm not sure this is expected behavior.

Imaging a scenario where changing one form control changes the options in this select input. If the user had previously selected a now deleted option, the input should be UNSET (selectIndex = -1) so that the user must choose again. However, this code updates the choice to one the user didn't make.

Reply:
I think I was trying to emulate default browser behavior where if you render a `select` the first option gets selected by default, but I like the idea of forcing `selectedIndex` to `-1` when no option is explicitly selected. I updated the code.